torchao/float8/config.py

```diff
@@ -62,6 +62,7 @@ class CastConfig:
     scaling_type: ScalingType = ScalingType.DYNAMIC
     scaling_granularity: ScalingGranularity = ScalingGranularity.TENSORWISE
     static_scale: Optional[torch.Tensor] = None
+    dtype: Optional[torch.dtype] = None
 
     def short_str(self):
```

Comment:
can we also add the dtype here, so it appears when we print an instance of `Float8Linear`?  `Float8Linear.__extra_repr__` calls this method.